packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,216 @@
+      expect(vaultAFterDFIAmt - vaultBeforeDFIAmt).toStrictEqual(10000)
+    }
+
+    {
+      const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+      const vaultBeforeBTCAcc = vaultBefore.collateralAmounts.length > 0
+        ? vaultBefore.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'BTC')
+        : undefined
+      const vaultBeforeBTCAmt = vaultBeforeBTCAcc !== undefined ? Number(vaultBeforeBTCAcc.split('@')[0]) : 0
+
+      const depositId = await tGroup.get(0).rpc.loan.depositToVault({
+        id: vaultId, from: collateralAddress, amount: '1@BTC'
+      })
+      expect(typeof depositId).toStrictEqual('string')
+      await tGroup.get(0).generate(1)
+      await tGroup.waitForSync()
+
+      const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])
+      const vaultAfterBTCAcc = vaultAfter.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'BTC')
+      const vaultAFterBTCAmt = Number(vaultAfterBTCAcc.split('@')[0])
+      expect(vaultAFterBTCAmt - vaultBeforeBTCAmt).toStrictEqual(1)
+    }
+  })
+
+  it('should depositToVault with utxos', async () => {
+    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
+    const vaultBeforeDFIAcc = vaultBefore.collateralAmounts.length > 0
+      ? vaultBefore.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'DFI')
+      : undefined
+    const vaultBeforeDFIAmt = vaultBeforeDFIAcc !== undefined ? Number(vaultBeforeDFIAcc.split('@')[0]) : 0
+
+    const utxo = await tGroup.get(0).container.fundAddress(collateralAddress, 250)
+    const depositId = await tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId, from: collateralAddress, amount: '250@DFI'
+    }, [utxo])
+    expect(typeof depositId).toStrictEqual('string')
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])
+    const vaultAfterDFIAcc = vaultAfter.collateralAmounts.find((amt: string) => amt.split('@')[1] === 'DFI')
+    const vaultAFterDFIAmt = Number(vaultAfterDFIAcc.split('@')[0])
+    expect(vaultAFterDFIAmt - vaultBeforeDFIAmt).toStrictEqual(250)
+
+    const rawtx = await tGroup.get(0).container.call('getrawtransaction', [depositId, true])
+    expect(rawtx.vin[0].txid).toStrictEqual(utxo.txid)
+    expect(rawtx.vin[0].vout).toStrictEqual(utxo.vout)
+  })
+
+  it('should be failed as vault must contain min 50% of DFI', async () => {
+    await tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId1, from: collateralAddress, amount: '100@DFI'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      id: vaultId1, from: collateralAddress, amount: '100@BTC'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('At least 50% of the vault must be in DFI')
```

Comment:
Could you add the calculation as a comment here? This is not straightforward for me to know that the DFI is not more than 50%